instaloader/structures.py

```diff
@@ -250,26 +250,32 @@ def typename(self) -> str:
         """Type of post, GraphImage, GraphVideo or GraphSidecar"""
         return self._field('__typename')
 
-    def get_sidecar_nodes(self) -> Iterator[PostSidecarNode]:
+    def get_sidecar_nodes(self, start=0, end=-1) -> Iterator[PostSidecarNode]:
         """Sidecar nodes of a Post with typename==GraphSidecar."""
         if self.typename == 'GraphSidecar':
             edges = self._field('edge_sidecar_to_children', 'edges')
             if any(edge['node']['is_video'] for edge in edges):
                 # video_url is only present in full metadata, issue #558.
                 edges = self._full_metadata['edge_sidecar_to_children']['edges']
+            if end < 0:
+                end = len(edges)
+            if start < 0:
+                start = len(edges)
```

Comment:
```suggestion
            if end < 0:
                end = len(edges) - 1
            if start < 0:
                start = len(edges) - 1
```

(alternatively `%= len(edges)`)